### PR TITLE
Organize KPrepublic, K.T.E.C, xiudi boards into directories

### PR DESCRIPTION
**Pull request by peepeetee**
## Description
~~Moved all KPrepublic board directories into the same directory~~

Organize KPrepublic, K.T.E.C, xiudi boards into folders.
The original scope of this pr was to organize kprepublic boards into a folder. As I was doing that, I erroneously included xiudi and K.T.E.C boards, as they were sold by KP republic. Thus the scope expanded to organize all of these into their respective directories, as there was a significant amount of xd boards.
List of affected boards:

Now in /kprepublic

- cospad
- bm16a
- bm16s
- bm40hsrgb
- bm43a
- bm60poker
- bm60rgb
- bm60rgb_iso
- bm68rgb
- jj4x4
- jj40
- jj50

Added 2021-08-06:
-bm65iso

Now in /xiudi

- xd002
- xd004
- xd60
- xd68
- xd75
- xd84
- xd87
- xd84pro
- xd96

Now in /ktec

- daisy
- staryu
- ergodone

All readmes has been updated to reflect the updated path

 Command used:
`mkdir kprepublic xiudi ktec`
`while read -r LINE; do mkdir kprepublic/"$LINE"; git mv "$LINE"/* kprepublic/"$LINE"; done < ~/list1`
`while read -r LINE; do mkdir xiudi/"$LINE"; git mv "$LINE"/* xiudi/"$LINE"; done < ~/list2`
`while read -r LINE; do mkdir ktec/"$LINE"; git mv "$LINE"/* ktec/"$LINE"; done < ~/list3`
## Types of Changes
Directory structure reorganization

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

*  #12155 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by peepeetee** [comment]
Please check if daisy, staryu, cospad and the xd series should be in this folder

**Comment by peepeetee** [comment]
@rgoulter @mechmerlin @ipetepete @markva @westfoxtrot @zvecr  @fougner

**Comment by peepeetee** [comment]
status: qmk link fails for keyboards with revisions; going to sleep

**Comment by peepeetee** [comment]
I did not delete the original folders, please advise

**Comment by fougner** [comment]
I don't have anything against the intention of the patch, xd87 is a kprepublic kbd so it makes sense to have it under a kprepublic dir. However, i do have some suggestions for the PR in general before I'd +1 it:

- do this as one single move so git correctly tracks file history etc. Currently the commit says "moved" but you're in fact copying files.
- the main keyboard dir seem to currently limit itself to lowercase, I assume we want to follow the same pattern.
- don't use past tense in commit messages, use imperative form as if to describe what the patch does _if_ applied, e.g. "move zap into zup".

**Comment by peepeetee** [comment]
so do a git mv instead of just copying, got it

File: keyboards/kprepublic/bm40hsrgb/readme.md [hunk 1]
```diff
@@ -12,10 +12,10 @@ This firmware was originally taken from a kbfirmware.json and manually converted
 
 Make example for this keyboard (after setting up your build environment):
 
-    make bm40hsrgb:default
+    make kprepublic/bm40hsrgb:default
```
Comment (filterpaper) [inline keyboards/kprepublic/bm40hsrgb/readme.md:15]:
```suggestion
    qmk compile -kb kprepublic/bm40hsrgb -km default
```
Perhaps you can also take this opportunity to update the build command documentation to use the `qmk` python script that is agnostic to current folder.

File: keyboards/kprepublic/bm40hsrgb/readme.md [hunk 2]
```diff
@@ -12,10 +12,10 @@ This firmware was originally taken from a kbfirmware.json and manually converted
...
 
 Flashing example for this keyboard:
 
-    make bm40hsrgb:default:flash
+    make kprepublic/bm40hsrgb:default:flash
```
Comment (filterpaper) [inline keyboards/kprepublic/bm40hsrgb/readme.md:19]:
```suggestion
    qmk flash -kb kprepublic/bm40hsrgb -km default
```

File: keyboards/kprepublic/bm43a/readme.md [hunk 3]
```diff
@@ -11,6 +11,6 @@ Hardware Availability: [KPRepublic](https://kprepublic.com/products/bm43a-bm43-4
 
 Make example for this keyboard (after setting up your build environment):
 
-    make bm43a:default
+    make kprepublic/bm43a:default
```
Comment (filterpaper) [inline keyboards/kprepublic/bm43a/readme.md:14]:
```suggestion
    qmk compile -kb kprepublic/bm43a -km default
```

File: keyboards/kprepublic/bm16s/readme.md [hunk 4]
```diff
@@ -8,6 +8,6 @@ Hardware Availability: [KPrepublic](https://kprepublic.com/collections/pcb/produ
 
 Make example for this keyboard (after setting up your build environment):
 
-    make bm16s:default
+    make kprepublic/bm16s:default
```
Comment (filterpaper) [inline keyboards/kprepublic/bm16s/readme.md:11]:
```suggestion
    qmk compile -kb kprepublic/bm16s -km default
```

File: keyboards/kprepublic/bm60rgb/readme.md [hunk 5]
```diff
@@ -8,6 +8,6 @@ A 60% hotswap inswitch RGB keyboard from KP Republic.
 
 Make example for this keyboard (after setting up your build environment):
 
-    make bm60rgb:default
+    make kprepublic/bm60rgb:default
```
Comment (filterpaper) [inline keyboards/kprepublic/bm60rgb/readme.md:11]:
```suggestion
    qmk compile -kb kprepublic/bm60rgb -km default
```

**Review by filterpaper — CHANGES_REQUESTED** [review]
Recommend documenting the use of `qmk` script as the default build command.

**Review by drashna — APPROVED** [review]

**Comment by peepeetee** [comment]
- Please review if Staryu should be included

- Considering renaming cospad to xd24 for better sorting, any inputs on that idea?

**Comment by peepeetee** [comment]
@drashna outdated due to scope expansion, please rescind approval

**Comment by peepeetee** [comment]
Please make sure that there are no other kprepublic, xiudi, or ktec boards

**Comment by peepeetee** [comment]
@yuhe00

**Comment by peepeetee** [comment]
Hi, can I get some advice on resolving merge conflicts? Would a rebase be prudent? 
Also apparently this pull request still needs to be reviewed but have no assigned reviewers, help

**Comment by drashna** [comment]
> Hi, can I get some advice on resolving merge conflicts? Would a rebase be prudent?
> Also apparently this pull request still needs to be reviewed but have no assigned reviewers, help

Yeah, you'd need to rebase, and then address any/all merge conflicts that arise.

**Comment by peepeetee** [comment]
@colinta I'm changing the readmes and has gotten to yours, thought you should be copied on this

**Comment by peepeetee** [comment]
Please check if staryu should be in /ktec

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
There are a bunch of Merge conflicts here.

**Review by colinta — APPROVED** [review]

**Comment by peepeetee** [comment]
I'll get to updating the readmes if this pr ended up being considered this cycle, at this point(f395c73) the only change is moving the sub-directories into directories

**Comment by peepeetee** [comment]
Please check if there are new keyboards added between the last time this PR is active and now

**Comment by peepeetee** [comment]
I see that the following KPrepublic keyboards are in the works, and that this is a duplicate of the now closed #5206 (ha this has history come help out with this PR @zvecr)


bm65rgb:
 #13361 by @bytesapart, 
BM60EC:
 #12253, https://github.com/umbacos/qmk_firmware/tree/master/keyboards/bm60ec_rgb by @umbacos,
bm80:
my own #12948,

We should coordinate and see if we can make these keyboards either 1. be added to this PR, or 2. be delayed after this PR so they can be sorted by the getgo

**Comment by peepeetee** [comment]
image
From #5206

In this PR I have a /kprepublic and a /ktec, what should I change?

**Review by noroadsleft — CHANGES_REQUESTED** [review]
One more file needs to be moved, and I think this is good to go.

`git mv keyboards/xd002/.noci keyboards/xiudi/xd002/.noci`

**Comment by umbacos** [comment]
Yes!

Il dom 8 ago 2021, 04:13 James Young ***@***.***> ha scritto:

**Comment by peepeetee** [comment]
> One more file needs to be moved, and I think this is good to go.
> 
> `git mv keyboards/xd002/.noci keyboards/xiudi/xd002/.noci`
> 

Done, any tips on how to include that in the commands I had in the description?

**Comment by noroadsleft** [comment]
I created a new branch then ran a series of `for` loops in QMK MSYS:

```
# Fetch the state of Pull Request 12159, and save it as a new branch `pr/12159`:
git fetch upstream +pull/12159/head:pr/12159;

# Create a new branch `move/kprepublic` from `develop`:
git checkout -b move/kprepublic develop

# change directory:
cd keyboards/

# create the subfolders:
mkdir -p kprepublic/ xiudi/ ktec/

# checking my git mv syntax; -n means "don't move or rename the files, but tell me what would be moved/renamed"
git mv -n cospad/ kprepublic/

# actually move the folders:
#   ... into kprepublic/ ...
for kb in cospad bm16a bm16s bm40hsrgb bm43a bm60poker bm60rgb bm60rgb_iso bm68rgb jj4x4 jj40 jj50; do git mv ${kb} kprepublic/; done
#   ... into xiudi/ ...
for kb in xd002 xd004 xd60 xd68 xd75 xd84 xd87 xd84pro xd96; do git mv ${kb} xiudi/; done
#   ... and into ktec/ ...
for kb in daisy staryu ergodone; do git mv ${kb} ktec/; done

# commit the moves:
git commit -m "move boards"

# compare this commit against your pull request:
git diff HEAD pr/12159
```

The diff output then shows me the difference between my changeset and yours, which showed the following:

1. I had forgotten to include `bm65iso` in my list of keyboards to place in the `kprepublic` folder – notice it's missing from the list of keyboards in the first `for` command.
2. You hadn't moved `keyboards/xd002/.noci` in your branch.

Dotfiles aren't special to Git; [they are a Linux convention](https://en.wikipedia.org/wiki/Hidden_file_and_hidden_directory#Unix_and_Unix-like_environments).

**Review by noroadsleft — APPROVED** [review]

**Comment by fauxpark** [comment]
That's not really how Git works. Regardless, the history was never lost:
image

**Comment by fauxpark** [comment]
Ah, I guess it's a feature of [Refined Github](https://github.com/refined-github/refined-github).

**Pull request merged**